Add norwegian-public-transport extension

rosvik opened the pull request:
Hi there 👋 

I've been working on [this extension](https://github.com/rosvik/raycast-public-transport) on and off for a few months now, and after testing it internally at my company, I thought it has become polished enough to share it on the store!

## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

It's an extension for viewing public transport info in and around Norway, based on [open data from Entur](https://developer.entur.org/pages-intro-setup-and-access#licenses), the official provider for such data in Norway (including busses, trains, ferries and more).

I've added functionality for looking up stops, and viewing upcoming departures for these stops. There is an option to save stops and departures as favorites for quick access.

More detailed info on features can be found in the README.

If you notice any issues, or have questions, please let me know ✌️

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

https://github.com/raycast/extensions/assets/1774972/4c7443f4-938f-4ff2-9b13-de04e181d8a6

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/norwegian-public-transport/src/Search/Actions.tsx:
```diff
@@ -0,0 +1,51 @@
+      <Action
+        title={
+          isFavorite
+            ? `Remove ${venue.properties.name} From Favorites`
```
```suggestion
            ? `Remove ${venue.properties.name} from Favorites`
```

pernielsentikaer commented on extensions/norwegian-public-transport/src/Search/Actions.tsx:
```diff
@@ -0,0 +1,51 @@
+            : `Add ${venue.properties.name} to Favorites`
+        }
+        icon={isFavorite ? Icon.StarDisabled : Icon.Star}
+        shortcut={
+          isFavorite ? { modifiers: ["cmd", "shift"], key: "s" } : { modifiers: ["cmd"], key: "s" }
+        }
```
```suggestion
        shortcut={Keyboard.Shortcut.Common.Pin}
```
Maybe try to use the same hotkeys as Raycast does, you can find them in `Keyboard`

rosvik replied:
Oh, sweet! I didn't realize there are defaults for this 💯

pernielsentikaer commented on extensions/norwegian-public-transport/src/Search/Actions.tsx:
```diff
@@ -0,0 +1,51 @@
+import { Action, ActionPanel, Icon, Image } from "@raycast/api";
```
```suggestion
import { Action, ActionPanel, Icon, Image, Keyboard } from "@raycast/api";
```

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could be nice with a custom List.EmptyView here 🙂

![raycast-norwegian-public-transport 2024-05-08 at 11 48 53](https://github.com/raycast/extensions/assets/38033107/39277543-1fba-4ca3-a161-09c6fefaf2be)

- Maybe also here if a departure don't have any lines

![raycast-norwegian-public-transport 2024-05-08 at 11 50 22](https://github.com/raycast/extensions/assets/38033107/0f0f8b51-30d6-4370-9055-19463205b8a6)



I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

rosvik commented:
@pernielsentikaer Thanks for the feedback! I've updated the extension with your suggestions ✨ 

Here are some screenshots of the `EmptyView`s I added:

![raycast-norwegian-public-transport 2024-05-11 at 15 51 30](https://github.com/raycast/extensions/assets/1774972/23e39285-a196-4dd4-b080-899f3811e178)
![raycast-norwegian-public-transport 2024-05-11 at 15 51 50](https://github.com/raycast/extensions/assets/1774972/26460afc-a78d-475c-8d4e-9acb990764e3)

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
